core/opts.cc

```diff
@@ -93,3 +93,19 @@ static bool ValidateMegabytesPerSocket(const char *, int32_t value) {
 DEFINE_int32(m, 1024, "Specifies how many megabytes to use per socket");
 static const bool _m_dummy[[maybe_unused]] =
     google::RegisterFlagValidator(&FLAGS_m, &ValidateMegabytesPerSocket);
+
+static bool ValidateBuffersPerSocket(const char *, int32_t value) {
+  if (value <= 0) {
+    LOG(ERROR) << "Invalid buffer count: " << value;
+    return false;
+  }
+  if (value % 512 != 0) {
```

Comment:
The comments in `packet.cc` suggest that the `rte_mempool_create` argument should be (2**n)-1 for some integral n, so perhaps we should check "is power of 2" here instead (so that dividing by 2 leaves inputs as a power of 2)? I'll leave more analysis to @sangjinhan but other than this, things seem OK here.

(Note that another option would be to round down to the nearest power of two in `packet.cc` after each divide-by-2.)

Reply:
Makes sense. I will post an update.